### PR TITLE
ET-4818 handle filter edge cases

### PR DESCRIPTION
**Pull request by janpetschexain**
**Reference**

- [ET-4818]
- requires #1026

**Summary**

- add tests for null & empty string array property filtering

### CODE REVIEW

File: web-api/tests/filter.rs [hunk 1]
```diff
@@ -842,6 +853,121 @@ fn test_filter_date() {
...
+            let documents = send_assert_json::<SemanticSearchResponse>(
+                &client,
+                client
+                    .post(personalization_url.join("/semantic_search")?)
+                    .json(&json!({
+                        "document": { "query": "zero" },
+                        "filter": { "p2": { "$eq": true } }
+                    }))
+                    .build()?,
+                StatusCode::OK,
```
Comment (rustonaut) [inline web-api/tests/filter.rs:941]: :art: maybe we add an annotation that it should fail once we validate against the schema?

**Review by rustonaut — APPROVED** [review]
lgtm
